datastore/gcloud/aio/datastore/datastore.py

```diff
@@ -37,7 +38,27 @@
 log = logging.getLogger(__name__)
 
 
-class Datastore:
+class DatastoreMeta(type):
```

Comment:
I think rather than doing this metaclass stuff we should stick with the solution we already have for now -- though this might be a better way to do it going forwards, I'd like to evaluate whether this is a worthwhile change on its own merits rather than at the same time as reviewing the `keyValue` logic. Note that this solution has some problems: it modifies the literal un-instantiated `Value` class for everyone, so as you called out that can cause several edge-cases.

Instead of doing this metaclass, let's give `Datastore`, `Entity`, and `Query` a `value_kind` entity to allow for custom subclassing. Should be the same as we have for all the other custom-subclass-able items; docs are [here](https://github.com/talkiq/gcloud-aio/tree/master/datastore#custom-subclasses). (and it would be great if you could update those docs with your changes!)

Reply:
Fair enough.
It's going to be a bit odd because it requires the client to specify a `value_kind` when they have a custom Key type. That also makes it a breaking change.